frontend/src/views/admin/matching/applicant-view/grid/grid-item/applicant-star.tsx

```diff
@@ -1,31 +1,22 @@
 import React from "react";
 import classNames from "classnames";
 import { MatchableAssignment } from "../../../types";
+import { useToggleStarred } from "../../../match-actions/modify-match-status";
 import { BsStarFill } from "react-icons/bs";
-import { upsertMatch } from "../../../actions";
-import { useThunkDispatch } from "../../../../../../libs/thunk-dispatch";
 
 /**
  * A button for toggling applicant's "starred" status for the currently-selected position.
  */
 export function ApplicantStar({ match }: { match: MatchableAssignment }) {
-    const dispatch = useThunkDispatch();
-    const updateStarred = React.useCallback(() => {
-        dispatch(
-            upsertMatch({
-                positionCode: match.position.position_code,
-                utorid: match.applicant.utorid,
-                starred: match.status !== "starred",
-            })
-        );
-    }, [dispatch, match]);
-
     return (
         <BsStarFill
             className={classNames("star-icon", {
                 filled: match.status === "starred",
             })}
-            onClick={updateStarred}
+            onClick={useToggleStarred(
```

Comment:
No. Hooks must be assigned to a variable at the top of the function and then used. Don't define them in component callbacks. If this changed to conditionally render something, the hook could randomly break.